### PR TITLE
Default metrics for kg and lbs

### PR DESCRIPTION
**Pull request by Malekai**

### CODE REVIEW

File: app/features/settings/application/SettingsApplicationActions.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
     type: PRESENT_END_SET_TIMER
 });
+
+
+export const presentSetMetric = () => ({
+	type: EDIT_DEFAULT_METRIC
+});
+
+
+
```
Comment (JonnyLinja) [inline app/features/settings/application/SettingsApplicationActions.js]: Remoe extra spaces

File: app/features/settings/application/SettingsApplicationPanel.js [hunk 2]
```diff
@@ -19,26 +20,43 @@ class SettingsApplicationPanel extends Component {
...
+                                  {this.props.defaultMetric}
+                                </Text>
+                            </TouchableOpacity>
+                        </View>
+                        <SettingsMetric />                        
                     </View>
-                    <SettingsEndSetTimerScreen />
-                </View>
             );
         } else {
```
Comment (JonnyLinja) [inline app/features/settings/application/SettingsApplicationPanel.js]: Make it work for Android

File: app/redux/reducers/SetsReducer.js [hunk 3]
```diff
@@ -93,6 +96,27 @@ const createDefaultState = () => {
...
+
+const setDefaultMetric = (state, action) => {
+    let newWorkoutData = state.workoutData.slice(0);
+    let latestSet = newWorkoutData[newWorkoutData.length - 1];
+    let setIndex = newWorkoutData.findIndex( set => set.setID === action.setID );
+    let set = newWorkoutData[setIndex];
+
+    let changes = {};
+    
+    if (!set.exercise && !set.weight && !set.rpe) {
```
Comment (JonnyLinja) [inline app/redux/reducers/SetsReducer.js]: Check against tags and videos and reps

File: app/redux/shared_actions/SetsActionCreators.js [hunk 4]
```diff
@@ -81,4 +94,6 @@ export const finishedUploadingSets = (revision) => ({
     revision: revision
 });
 
-export const failedUploadSets = () => ({ type: FAILED_UPLOAD_SETS });
\ No newline at end of file
+export const failedUploadSets = () => ({ type: FAILED_UPLOAD_SETS });
+
+
```
Comment (JonnyLinja) [inline app/redux/shared_actions/SetsActionCreators.js]: Remove extra space

File: app/redux/reducers/SetsReducer.js [hunk 5]
```diff
@@ -93,6 +97,27 @@ const createDefaultState = () => {
...
+
+const setDefaultMetric = (state, action) => {
+    let newWorkoutData = state.workoutData.slice(0);
+    let latestSet = newWorkoutData[newWorkoutData.length - 1];
+    let setIndex = newWorkoutData.findIndex( set => set.setID === action.setID );
+    let set = newWorkoutData[setIndex];
+
+    let changes = {};
+    
+    if (!set.exercise && !set.weight && !set.rpe && !set.tags && !set.reps && !set.tags && !set.videoFileURL) {
```
Comment (JonnyLinja) [inline app/redux/reducers/SetsReducer.js]: either move this to its own function or add a comment above it explaining what this if statement is trying to do

File: app/features/settings/application/SettingsApplicationPanel.js [hunk 6]
```diff
@@ -19,26 +20,43 @@ class SettingsApplicationPanel extends Component {
...
+                        </View>
+                        <SettingsEndSetTimerScreen />
+                        <View>
+                            <Text style={ SETTINGS_PANEL_STYLES.headerText }>
+                                Set default metric
+                            </Text>
+                        </View>
+                        <View>
+                            <TouchableOpacity onPress={() => this._tappedSetMetric()}>
+                                <Text style={SETTINGS_PANEL_STYLES.tappableText  }>
```
Comment (JonnyLinja) [inline app/features/settings/application/SettingsApplicationPanel.js]: Remove extra spaces

File: app/ActionTypes.js [hunk 7]
```diff
@@ -82,6 +82,9 @@ export const SAVE_HISTORY_VIDEO = 'SAVE_HISTORY_VIDEO';
 export const SAVE_END_SET_TIMER = 'SAVE_END_SET_TIMER';
 export const PRESENT_END_SET_TIMER = 'PRESENT_END_SET_TIMER';
 export const DISMISS_END_SET_TIMER = 'DISMISS_END_SET_TIMER';
+export const SET_DEFAULT_METRIC = 'SET_DEFAULT_METRIC';
+export const EDIT_DEFAULT_METRIC = 'EDIT_SET_METRIC';
```
Comment (JonnyLinja) [inline app/ActionTypes.js]: Rename EDIT_DEFAULT_METRIC to PRESENT_DEFAULT_METRIC

File: app/ActionTypes.js [hunk 8]
```diff
@@ -82,6 +82,9 @@ export const SAVE_HISTORY_VIDEO = 'SAVE_HISTORY_VIDEO';
 export const SAVE_END_SET_TIMER = 'SAVE_END_SET_TIMER';
 export const PRESENT_END_SET_TIMER = 'PRESENT_END_SET_TIMER';
 export const DISMISS_END_SET_TIMER = 'DISMISS_END_SET_TIMER';
+export const SET_DEFAULT_METRIC = 'SET_DEFAULT_METRIC';
```
Comment (JonnyLinja) [inline app/ActionTypes.js]: Rename SET_DEFAULT_METRIC to SAVE_DEFAULT_METRIC

File: app/ActionTypes.js [hunk 9]
```diff
@@ -82,6 +82,9 @@ export const SAVE_HISTORY_VIDEO = 'SAVE_HISTORY_VIDEO';
...
+export const EDIT_DEFAULT_METRIC = 'EDIT_SET_METRIC';
+export const DISMISS_SET_METRIC = 'DISMISS_SET_METRIC';
```
Comment (JonnyLinja) [inline app/ActionTypes.js]: Rename DISMISS_SET_METRIC to DISMISS_DEFAULT_METRIC

File: app/redux/shared_actions/SettingsActionCreators.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+// These exist as a shared action creator because saving values can be done from all over the app
+
+import { 
+	SAVE_END_SET_TIMER, 
+	SET_DEFAULT_METRIC
+} from 'app/ActionTypes';
+
+import * as SetsSelectors from 'app/redux/selectors/SetsSelectors';
+
+export const setDefaultMetric = (metric = 'kgs') => (dispatch, getState) => {
```
Comment (JonnyLinja) [inline app/redux/shared_actions/SettingsActionCreators.js]: rename to setDefaultMetric to saveDefaultMetric

File: app/redux/reducers/SettingsReducer.js [hunk 11]
```diff
@@ -3,17 +3,34 @@ import {
...
-    UPDATE_SET_DATA_FROM_SERVER
+    UPDATE_SET_DATA_FROM_SERVER,
+    SET_DEFAULT_METRIC,
+    EDIT_DEFAULT_METRIC,
+    DISMISS_SET_METRIC
 } from 'app/ActionTypes';
 
 const defaultState = {
+    defaultMetric: 'kgs',
+    editingDefaultMetric: false,
```
Comment (JonnyLinja) [inline app/redux/reducers/SettingsReducer.js]: rename editingDefaultMetric to isEditingDefaultMetric

File: app/features/workout/WorkoutScreen.js [hunk 12]
```diff
@@ -4,10 +4,10 @@ import { connect } from 'react-redux';
 import * as SetsSelectors from 'app/redux/selectors/SetsSelectors';
 import * as DateUtils from 'app/utility/transforms/DateUtils';
 import * as RepDataMap from 'app/utility/transforms/RepDataMap';
-import * as SetTimeCalculator from 'app/utility/transforms/SetTimeCalculator';
```
Comment (JonnyLinja) [inline app/features/workout/WorkoutScreen.js]: Add SetTimeCalculator back in for this entire file.

Rebase fucked up somehow. Need SetTimeCalculator or else shit will break.

File: app/features/workout/WorkoutScreen.js [hunk 13]
```diff
@@ -36,14 +36,15 @@ const createViewModels = (sets) => {
...
+
         } else if (!set.removed) {
             if (lastExerciseName !== null && lastExerciseName === set.exercise) {
                 setNumber++;
             } else {
                 setNumber = 1;
             }
         }
-        array.push(createHeaderViewModel(set, setNumber, lastExerciseName));
+        array.push(createHeaderViewModel(set, setNumber));
```
Comment (JonnyLinja) [inline app/features/workout/WorkoutScreen.js]: Add the lastExerciseName back too

File: app/features/workout/WorkoutList.js [hunk 14]
```diff
@@ -61,7 +61,6 @@ class WorkoutList extends Component {
                                 weight={item.weight}
                                 metric={item.metric}
                                 rpe={item.rpe}
-                                bias={item.bias}
```
Comment (JonnyLinja) [inline app/features/workout/WorkoutList.js]: Re-add bias on WorkoutList

File: app/features/settings/application/metric/SettingsMetric.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import PickerModal from 'app/shared_features/picker/PickerModal';
+import * as Actions from './SettingsMetricActions';
+
+
```
Comment (JonnyLinja) [inline app/features/settings/application/metric/SettingsMetric.js]: Remove extra space

File: app/features/settings/application/metric/SettingsMetric.js [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+const mapStateToProps = (state) => ({
+    modalShowing: state.settings.editingDefaultMetric,
```
Comment (JonnyLinja) [inline app/features/settings/application/metric/SettingsMetric.js]: rename modalShowing to isModalShowing

Do this for everywhere modalShowing is.

File: app/redux/reducers/SetsReducer.js [hunk 17]
```diff
@@ -93,6 +97,27 @@ const createDefaultState = () => {
     };
 };
 
+// Set default metric
+
+const setDefaultMetric = (state, action) => {
```
Comment (JonnyLinja) [inline app/redux/reducers/SetsReducer.js]: Rename the function and the comment to saveDefaultMetric

File: app/features/settings/application/SettingsApplicationActions.js (hunk 1 above)
Comment (Malekai) [reply app/features/settings/application/SettingsApplicationActions.js]: Done

File: app/redux/reducers/SetsReducer.js (hunk 5 above)
Comment (Malekai) [reply app/redux/reducers/SetsReducer.js]: Done

File: app/features/settings/application/SettingsApplicationPanel.js (hunk 6 above)
Comment (Malekai) [reply app/features/settings/application/SettingsApplicationPanel.js]: Done

File: app/ActionTypes.js (hunk 8 above)
Comment (Malekai) [reply app/ActionTypes.js]: Done

File: app/ActionTypes.js (hunk 7 above)
Comment (Malekai) [reply app/ActionTypes.js]: Done

File: app/redux/shared_actions/SettingsActionCreators.js (hunk 10 above)
Comment (Malekai) [reply app/redux/shared_actions/SettingsActionCreators.js]: Done

File: app/features/workout/WorkoutScreen.js (hunk 12 above)
Comment (Malekai) [reply app/features/workout/WorkoutScreen.js]: Done

File: app/features/workout/WorkoutList.js (hunk 14 above)
Comment (Malekai) [reply app/features/workout/WorkoutList.js]: Done

File: app/features/workout/WorkoutScreen.js (hunk 13 above)
Comment (Malekai) [reply app/features/workout/WorkoutScreen.js]: Done

File: app/features/workout/WorkoutScreen.js (hunk 12 above)
Comment (Malekai) [reply app/features/workout/WorkoutScreen.js]: Done

File: app/redux/reducers/SettingsReducer.js (hunk 11 above)
Comment (Malekai) [reply app/redux/reducers/SettingsReducer.js]: Done

File: app/features/settings/application/metric/SettingsMetric.js (hunk 15 above)
Comment (Malekai) [reply app/features/settings/application/metric/SettingsMetric.js]: Done

File: app/features/settings/application/metric/SettingsMetric.js (hunk 16 above)
Comment (Malekai) [reply app/features/settings/application/metric/SettingsMetric.js]: Done

File: app/redux/reducers/SetsReducer.js (hunk 17 above)
Comment (Malekai) [reply app/redux/reducers/SetsReducer.js]: Done

File: app/features/settings/application/SettingsApplicationPanel.js [hunk 18]
```diff
@@ -19,26 +20,43 @@ class SettingsApplicationPanel extends Component {
         this.props.tapEndSetTimer()
     }
 
+    _tappedSaveMetric() {
+        this.props.tapSaveDefaultMetric()
```
Comment (JonnyLinja) [inline app/features/settings/application/SettingsApplicationPanel.js]: rename to tapDefaultMetric

File: app/redux/reducers/SettingsReducer.js [hunk 19]
```diff
@@ -3,17 +3,34 @@ import {
...
-    UPDATE_SET_DATA_FROM_SERVER
+    UPDATE_SET_DATA_FROM_SERVER,
+    SAVE_DEFAULT_METRIC,
+    PRESENT_DEFAULT_METRIC,
+    DISMISS_SET_METRIC
 } from 'app/ActionTypes';
 
 const defaultState = {
+    defaultMetric: 'kgs',
+    iseditingDefaultMetric: false,
```
Comment (JonnyLinja) [inline app/redux/reducers/SettingsReducer.js]: isEditingDefaultMetric

File: app/features/workout/WorkoutScreen.js [hunk 20]
```diff
@@ -165,7 +167,7 @@ const createRowViewModels = (set) => {
 };
 
 const createFooterVM = (set, lastSetEndTime) => {
-    let restInMS = new Date(SetTimeCalculator.startTime(set)) - new Date(lastSetEndTime);
+    let restInMS = new Date(set.startTime) - new Date(lastSetEndTime);
     let footerVM = {
         type: "footer",
```
Comment (JonnyLinja) [inline app/features/workout/WorkoutScreen.js]: restore all SetTimeCalculator

File: app/redux/reducers/SetsReducer.js [hunk 21]
```diff
@@ -93,6 +97,28 @@ const createDefaultState = () => {
...
+const saveDefaultMetric = (state, action) => {
+    let newWorkoutData = state.workoutData.slice(0);
+    let latestSet = newWorkoutData[newWorkoutData.length - 1];
+    let setIndex = newWorkoutData.findIndex( set => set.setID === action.setID );
+    let set = newWorkoutData[setIndex];
+
+    let changes = {};
+    
+    // Check if set is empty before allowing metric to change
+    if (!set.exercise && !set.weight && !set.rpe && !set.tags && !set.reps && !set.tags && !set.videoFileURL) {
```
Comment (JonnyLinja) [inline app/redux/reducers/SetsReducer.js]: do tags only once

tags and reps should check length AND not empty state

File: app/redux/reducers/SettingsReducer.js (hunk 19 above)
Comment (Malekai) [reply app/redux/reducers/SettingsReducer.js]: Done

File: app/features/workout/WorkoutScreen.js (hunk 20 above)
Comment (Malekai) [reply app/features/workout/WorkoutScreen.js]: Done

File: app/redux/reducers/SetsReducer.js (hunk 21 above)
Comment (Malekai) [reply app/redux/reducers/SetsReducer.js]: Done

File: app/features/settings/application/SettingsApplicationPanel.js (hunk 18 above)
Comment (Malekai) [reply app/features/settings/application/SettingsApplicationPanel.js]: Done

File: app/features/settings/application/SettingsApplicationPanel.js (hunk 2 above)
Comment (Malekai) [reply app/features/settings/application/SettingsApplicationPanel.js]: Done

File: app/features/settings/application/SettingsApplicationPanel.js [hunk 22]
```diff
@@ -19,26 +20,43 @@ class SettingsApplicationPanel extends Component {
         this.props.tapEndSetTimer()
     }
 
+    _tapDefaultMetric() {
```
Comment (JonnyLinja) [inline app/features/settings/application/SettingsApplicationPanel.js]: tappedDefaultMetric

File: app/features/settings/application/SettingsApplicationPanel.js [hunk 23]
```diff
@@ -19,26 +20,43 @@ class SettingsApplicationPanel extends Component {
...
+        this.props.tapSaveDefaultMetric()
```
Comment (JonnyLinja) [inline app/features/settings/application/SettingsApplicationPanel.js]: tapDefaultMetric

File: app/features/settings/application/metric/SettingsMetric.js [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+import PickerModal from 'app/shared_features/picker/PickerModal';
+import * as Actions from './SettingsMetricActions';
+
+const mapStateToProps = (state) => ({
+    isModalShowing: state.settings.isEditingDefaultMetric,
+    items: [
+        {label: 'kgs', value: 'kgs'},
+        {label: 'lbs', value: 'lbs'}
+    ],
+    selectedValue: state.settings.defaultMetric
```
Comment (JonnyLinja) [inline app/features/settings/application/metric/SettingsMetric.js]: Selectors!

**Pull request merged**